cluster/swarm/cluster.go

```diff
@@ -119,7 +119,7 @@ func (c *Cluster) UnregisterEventHandler(h cluster.EventHandler) {
 func (c *Cluster) generateUniqueID() string {
 	for {
 		id := stringid.GenerateRandomID()
-		if c.Container(id) == nil {
+		if _, err := c.Container(id); err != nil {
```

Comment:
In your change when there are more than 1 container with this ID, it returns error. So your logic here is not right.